source/FreeRTOS_IP_Utils.c

```diff
@@ -220,13 +224,21 @@ static NetworkBufferDescriptor_t * prvPacketBuffer_to_NetworkBuffer( const void
     else
     {
         /* Obtain the network buffer from the zero copy pointer. */
-        uxBuffer = ipPOINTER_CAST( uintptr_t, pvBuffer );
+
+        /* This conversion between a pointer to void and 'uintptr_t' is being
+         * suppressed as 'uintptr_t' is guaranteed to be a pointer size on the
+         * used platform.  */
+        /* coverity[misra_c_2012_rule_11_1_violation] */
+        uxBuffer = ( uintptr_t ) pvBuffer;
 
         /* The input here is a pointer to a packet buffer plus some offset.  Subtract
          * this offset, and also the size of the header in the network buffer, usually
          * 8 + 2 bytes. */
         uxBuffer -= ( uxOffset + ipBUFFER_PADDING );
 
+        uint8_t * pcBufferOffset = ( uint8_t * ) pvBuffer;
+        pcBufferOffset = &( pcBufferOffset [ 0 - ( uxOffset + ipBUFFER_PADDING ) ] );
```

Comment:
I think you added the 2 lines for testing?
~~~c
    uint8_t * pcBufferOffset = ( uint8_t * ) pvBuffer;
    pcBufferOffset = &( pcBufferOffset [ 0 - ( uxOffset + ipBUFFER_PADDING ) ] );
~~~
I think that I prefer the earlier :
~~~c
    uxBuffer -= ( uxOffset + ipBUFFER_PADDING );
~~~
In `0 - ( uxOffset + ipBUFFER_PADDING )` there is a hidden cast from unsigned values to a negative signed number that serves as an index.